[FIX] mail: remove admin from mail.alias

yajo opened the pull request:
~~Since Administrator is not going to be used anymore, it is best to archive that partner, just like it is done with that user. If not doing so, accessing the user will display a warning about this.~~

When a mail.alias record belongs to admin, it will try to send mails in name of a user that most likely has wrong data in it, and is actually garbage. That can produce a problem where aliases create records but don't notify followers, or even worse, that same problem in loop (depending possibly on if you use IMAP4 or POP3 to fetch mails).

@Tecnativa TT24347

pedrobaeza commented:
@StefanRijnhart you did this part. Can you please check?

StefanRijnhart commented:
I don't understand why you say that the Administrator is not going to be used anymore. I believe it is, but its privileges will now be based on group membership instead of having a magic ID (that is the prerogative of the inactive root user).

StefanRijnhart commented on addons/mail/migrations/12.0.1.0/post-migration.py:
```diff
@@ -31,9 +31,25 @@ def fill_mail_notification_mail_id(cr):
+    """
+    openupgrade.logged_query(
+        cr,
+        "UPDATE mail_alias SET alias_user_id = NULL WHERE alias_user_id = 1",
```
Without this query, ownership of the aliases would be transferred to the (new) Admin user in the end migration. I see system aliases do not have an owner in new Odoo 12 databases but can have the Admin user in older databases so I think resetting the alias_user_id here is correct.

StefanRijnhart commented on odoo/addons/base/migrations/12.0.1.3/end-migration.py:
```diff
@@ -104,11 +104,11 @@ def fork_off_system_user(env):
-            SET active = FALSE WHERE id = %s """,
-        (partner_root.id,))
+            SET active = FALSE WHERE id IN %s """,
+        ((partner_root.id, partner_admin.id),))
```
I think setting the admin user's partner to inactive is incorrect. On new Odoo 12 databases there is an active admin user with an active partner.

yajo replied:
Oh I see. It's weird that's not just demo data. I guess this could have been a side effect of seeing a warning in the admin user form, but yes probably this isn't the real fix to the problem.

I'll revert this.

Commit:
[FIX] mail: remove admin from mail.alias

When a mail.alias record belongs to admin, it will try to send mails in name of a user that most likely has wrong data in it, and is actually garbage. That can produce a problem where aliases create records but don't notify followers, or even worse, that same problem in loop (depending possibly on if you use IMAP4 or POP3 to fetch mails).

@Tecnativa TT24347

yajo commented:
All solved.

StefanRijnhart reviewed (approved):
Thanks for the update!
